src/server/routes/obvius.js

```diff
@@ -112,17 +112,23 @@ function obviusLog(req, res, next){
  * Verifies an Obvius request via username and password.
  */
 function verifyObviusUser(req, res, next){
-	// First we ensure that the password and email parameters are provided.
-	if (!req.param('password')) {
+	// First we ensure that the password and username parameters are provided.
+	const password = req.param('password');
+	// allowing for backwards compatibility if previous eGuage meters are using the 'email' parameter instead of
+		// the 'username' parameter to login. Developers need to decide in the future if we should deprecate email
+		// or continue to allow this backwards compatibility
+	const username = req.param('username') || req.param('email');
+
+	if (!password) {
 		failure(req, res, 'password parameter is required.');
 		return;
-	} else if (!req.param('email')){
-		failure(req, res, 'email parameter is required.');
+	} else if (!username) {
+		failure(req, res, 'username parameter is required.');
 		return;
 	} else { // Authenticate Obvius user.
-		req.body.email = req.param('email');
-		req.body.password = req.param('password');
-		obviusEmailAndPasswordAuthMiddleware('Obvius pipeline')(req, res, next);
+		req.body.username = username;
+		req.body.password = password;
+		obviusUsernameAndPasswordAuthMiddleware('Obvius pipeline')(req, res, next);
```

Comment:
I'm not really knowledgeable here but I want to verify you did not break this code. The original set req.body.password and that is what src/server/routes/authenticator.js uses. Is the fact it is in req.param('password') the same? We have to be very careful with any authentication changes.

Reply:
I do not believe so. This: req.body.username = req.param('username'); I believe is equivalent to this: const username = req.param('username') || req.param('email'); req.body.username = username;

Reply:
Although, we do not have an obvius test correct? Is there a way we can create a test or is there one I am not seeing?